### PR TITLE
#1594 add scalper orderbook stop orders moving

### LINKED ISSUE #1594: [FEATURE] Перетаскивание заявок в стакане
### Опишите проблему, которую решает ваша фича

Здравствуйте. В веб-терминале Astras изменить выставленную заявку возможно только отменой старой и выставлением новой, либо через вкладку "Заявки" виджета "Блоттер". Оба способа в большинстве случаев занимают много времени. Возможно изменить перетаскиванием в виджете "Тех. анализ", но это не совсем удобно при работе в стакане или в случае, если виджет "Тех. анализ" вовсе не используется при работе (открыты виджеты "Скальперский стакан" и "График" по отслеживаемым инструментам.

### Опишите предлагаемое вами решение

Снимок экрана 2024-04-03 в 15 08 08

Сделайте отдельную кнопку для отмены заявки в стакане (_крестик чуть правее заявки_), а при нажатии на наименование заявки (_SL(1)_) и перетаскивание возможность изменять цену заявки. Проще говоря сделайте так, чтобы можно было перетаскивать заявки в стакане, а для отмены отдельную кнопку.

### Опишите альтернативные решения, о которых вы думали

Как альтернативный вариант для сохранения компактности можно сделать перемещение заявок левой кнопкой мыши, а отмену заявки правой кнопкой мыши.

### Дополнительная информация

Будет особенно круто, если сделаете оба варианта и добавите пункт в настройки виджета "Скальперский стакан", а там каждый выберет сам себе как ему удобнее.

### Правила

- [X] Я согласен следовать правилам

### PR DESCRIPTION
**Pull request by agushasok**
Fixes #1594 

# Description

add scalper orderbook stop orders moving

# Testing

add scalper orderbook widget
try replace stop order

# Risk

No major changes

# Do you agree with those?
- [] I agree to follow the [Contributing guidelines](https://github.com/alor-broker/Astras-Trading-UI/blob/master/CONTRIBUTING.md)
- [] I agree to follow the terms of the [Code of Conduct](https://github.com/alor-broker/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

File: src/app/modules/scalper-order-book/models/scalper-order-book.model.ts [hunk 1]
```diff
@@ -11,6 +12,8 @@ export interface CurrentOrderDisplay {
   side: Side;
   linkedPrice: number;
   displayVolume: number;
+  price?: number;
```
Comment (agushasok) [inline src/app/modules/scalper-order-book/models/scalper-order-book.model.ts:15]: переименовать в price и triggerPrice

File: src/app/modules/scalper-order-book/services/scalper-orders.service.ts [hunk 2]
```diff
@@ -381,26 +381,49 @@ export class ScalperOrdersService {
...
+              },
+              order.portfolio
+            ).subscribe();
+            break;
+          case "stoplimit":
+            this.orderService.submitStopLimitOrderEdit({
+                ...baseOrderEditData,
+                condition: order.condition!,
+                triggerPrice: updates.price,
+                price: order.price! - (order.linkedPrice - updates.price),
```
Comment (agushasok) [inline src/app/modules/scalper-order-book/services/scalper-orders.service.ts:407]: добавить округление

File: src/app/modules/scalper-order-book/components/scalper-order-book-table/scalper-order-book-table.component.ts [hunk 3]
```diff
@@ -176,8 +176,8 @@ export class ScalperOrderBookTableComponent implements OnInit {
     ]).pipe(
       map(([settings, body, displayRange, currentOrders, themeSettings]) => {
         const displayRows = body.slice(displayRange!.start, Math.min(displayRange!.end + 1, body.length));
-        const minOrderPrice = Math.min(...currentOrders.map(x => x.linkedPrice));
-        const maxOrderPrice = Math.max(...currentOrders.map(x => x.linkedPrice));
+        const minOrderPrice = Math.min(...currentOrders.map(x => x.triggerPrice ?? x.price!));
```
Comment (PilgrimViis) [inline src/app/modules/scalper-order-book/components/scalper-order-book-table/scalper-order-book-table.component.ts:179]: Пожалуйста убери ! и поставь ?? throw new Error("Price or trigger price should be present")

File: src/app/modules/scalper-order-book/components/scalper-order-book-body/scalper-order-book-body.component.ts [hunk 4]
```diff
@@ -261,8 +261,8 @@ export class ScalperOrderBookBodyComponent implements OnInit, AfterViewInit, OnD
           : null;
 
         return {
-          up: !!(upPrice ?? 0) && !!currentOrders.find(o => o.linkedPrice > upPrice!),
-          down: !!(downPrice ?? 0) && !!currentOrders.find(o => o.linkedPrice < downPrice!),
+          up: !!(upPrice ?? 0) && !!currentOrders.find(o => (o.triggerPrice ?? o.price!) > upPrice!),
+          down: !!(downPrice ?? 0) && !!currentOrders.find(o => (o.triggerPrice ?? o.price!) < downPrice!),
```
Comment (PilgrimViis) [inline src/app/modules/scalper-order-book/components/scalper-order-book-body/scalper-order-book-body.component.ts:265]: Пожалуйста убери ! и поставь ?? throw new Error("Price or trigger price should be present")

**Review by PilgrimViis — CHANGES_REQUESTED** [review]

**Pull request closed without merging**